test/timeevol/cluster_projectors.jl

```diff
@@ -0,0 +1,126 @@
+using Test
+using TensorKit
+using PEPSKit
+using LinearAlgebra
+using Random
+import MPSKitModels: hubbard_space
+using PEPSKit: sdiag_pow, _cluster_truncate!
+include("cluster_tools.jl")
+
+nrm = 20
+Vspaces = [
+    (ℂ^2, ℂ^4, (ℂ^12)'),
+    (
+        Vect[FermionParity](0 => 1, 1 => 1),
+        Vect[FermionParity](0 => 2, 1 => 2),
+        Vect[FermionParity](0 => 6, 1 => 6)',
+    ),
+]
+
+@testset "Cluster bond truncation with projectors" begin
+    Random.seed!(0)
+    N, n = 5, 2
+    for (Vphy, Vns, V) in Vspaces
+        Vvirs = fill(Vns, N + 1)
+        Vvirs[n + 1] = V
+        Ms1 = Vector{AbstractTensorMap}(undef, N)
+        for i in 1:N
+            Vw, Ve = Vvirs[i], Vvirs[i + 1]
+            Ms1[i] = rand(Vw ← Vphy' ⊗ Vns ⊗ Vns' ⊗ Ve) / nrm
+        end
```

Comment:
This is again not entirely type stable. I don't mind it as much here in the tests, but in general a `map` construction is probably better